libevmasm/SemanticInformation.h

```diff
@@ -45,6 +45,32 @@ struct SemanticInformation
 		Write
 	};
 
+	enum class Location { Storage, Memory };
+
+	/**
+	 * Represents a read or write operation from or to one of the data locations.
+	 */
+	struct Operation
+	{
+		Location location;
+		Effect effect;
+		/// Start of affected area as an index into the parameters.
+		/// Unknown if not provided.
+		std::optional<size_t> startParameter;
+		/// Length of the affected area as an index into the parameters (if this is an opcode).
+		/// Unknown if neither this nor lengthConstant is provided.
+		std::optional<size_t> lengthParameter;
+		/// Length as a constant.
+		/// Unknown if neither this nor lengthArgument is provided.
+		std::optional<size_t> lengthConstant;
```

Comment:
Since this and ``lengthParameter`` are mutually exclusive, I wonder if we should make this a variant, e.g.
```std::variant<Parameter, Constant, Unknown> length;`` - but I'm not sure if that'd make things more or less annoying on the usage end (especially since we need additional types encapsulating a simple ``size_t`` for that).

Reply:
Yeah, an optional of a variant would be best, espeacilly with a data enum, but we don't have that in C++...

Reply:
Well, defining ``struct Paramter { size_t value; };`` and ``struct Constant { size_t value; };`` and using them in a ``std::variant`` kind of does that...
Plus either the ``variant`` inside an ``optional``, or with an additional ``struct Unknown {};`` in the variant, or with ``std::monostate`` in the variant (i.e. "empty")...
But yeah, not sure that's really better in the end...